docs/reference/function-configuration/function-configuration-reference.md

```diff
@@ -56,35 +60,39 @@ The `spec` section contains the requirements and attributes and has the followin
 | description | string | A textual description of the function |
 | handler | string | The entry point to the function, in the form of `package:entrypoint`. Varies slightly between runtimes, see the appropriate runtime documentation for specifics |
 | runtime | string | The name of the language runtime. One of: `golang`, `python:2.7`, `python:3.6`, `shell`, `java`, `nodejs`, `pypy` | 
-| image | string | The container image holding the function |
-| env | map | A name-value environment-variable tuple. It is also possible to point to secrets, as demonstrated in the following example |
+| <a id="spec.image"></a>image | string | The name of the function's container image &mdash; used for the `image` [code-entry type](#spec.build.codeEntryType); see [Code-Entry Types](/docs/reference/function-configuration/code-entry-types.md#code-entry-type-image) |
+| env | map | A name-value environment-variables tuple; it's also possible to reference secrets from the map elements, as demonstrated in the [specifcation example](#spec-example) |
 | volumes | map | A map in an architecture similar to Kubernetes volumes, for Docker deployment |
 | replicas | int | The number of desired instances; 0 for auto-scaling. |
 | minReplicas | int | The minimum number of replicas |
 | maxReplicas | int | The maximum number of replicas |
 | targetCPU | int | Target CPU when auto scaling, as a percentage (default: 75%) |
 | dataBindings | See reference | A map of data sources used by the function ("data bindings") |
 | triggers.(name).maxWorkers | int | The max number of concurrent requests this trigger can process |
-| triggers.(name).kind | string | The kind of trigger. One of `http`, `kafka`, `kinesis`, `eventhub`, `cron`, `nats`, `rabbitmq` |
+| triggers.(name).kind | string | The trigger type (kind) - `cron` \| `eventhub` \| `http` \| `kafka` \| `kinesis` \| `nats` \| `rabbitmq` |
 | triggers.(name).url | string | The trigger specific URL (not used by all triggers) |
 | triggers.(name).annotations | list of strings | Annotations to be assigned to the trigger, if applicable |
 | triggers.(name).workerAvailabilityTimeoutMilliseconds | int | The number of milliseconds to wait for a worker if one is not available. 0 = never wait |
 | triggers.(name).attributes | See [reference](/docs/reference/triggers) | The per-trigger attributes |
-| build.path | string | A local directory or URL to a file/archive containing source and configuration |
-| build.functionSourceCode | string | The source code of the function, encoded in Base64. Mutually exclusive with build.path |
+| <a id="spec.build.path"></a>build.path | string | Function-code GitHub directory or archive-file path for the `github` or `archive` [code-entry type](#spec.build.codeEntryType); see [Code-Entry Types](/docs/reference/function-configuration/code-entry-types.md#external-source-code-entry-types) |
```

Comment:
1. About the github part - it is a URL for github repository
2. It can also be a path for a single code file - for example `https://www.host.com/my-python-func.py`

Reply:
Following our f2f discussions, I edited the description of the `spec.build.path` field in *function-configuration-reference.md* and I edited the *code-entry-types.md* reference to document also the source-code file code-entry type. As part of this change, I reorganized and retitled the existing doc sections and made related doc edits. See the diff for commit 1a81f34.